server/src/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -497,7 +498,7 @@ protected Long restart(final HaWorkVO work) {
 
                 boolean fenced = false;
                 if (alive == null) {
-                    s_logger.debug("Fencing off VM that we don't know the state of");
+                    s_logger.debug("Fencing off VM " + vm + " that we don't know the state of");
```

Comment:
In my view, the ``Supplier`` approach is just as complex, but in another way.  In the long term, we will be adopting SLF4J which supports string templates -- allowing constructs as such as:

```
LOG.debug("I am having a problem VM %s", vm.getId());
```

This approach defers string concatenation in a more straight forward manner.